src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Current.Manifest/WorkloadTelemetry.targets

```diff
@@ -0,0 +1,45 @@
+          AfterTargets="Publish"
+          DependsOnTargets="_WorkloadGatherTelemetryData">
+
+    <ItemGroup>
+      <WorkloadPublishProperties Include="TargetPlatformIdentifier" Value="$(TargetPlatformIdentifier)" />
+      <WorkloadPublishProperties Include="RuntimeIdentifier" Value="$(RuntimeIdentifier)" />
+      <WorkloadPublishProperties Include="UseMaui" Value="$(UseMaui)" />
+      <WorkloadPublishProperties Include="BlazorWasm" Value="$(_WorkloadUsesBlazorWasm)" />
+      <WorkloadPublishProperties Include="WasmSDK" Value="$(_WorkloadUsesWasmSDK)" />
+      <WorkloadPublishProperties Include="Avalonia" Value="$(_WorkloadUsesAvalonia)" />
+      <WorkloadPublishProperties Include="Uno" Value="$(_WorkloadUsesUno)" />
+      <WorkloadPublishProperties Include="MonoAOT" Value="$(_WorkloadUsesMonoAOT)" />
+      <WorkloadPublishProperties Include="NativeAOT" Value="$(_WorkloadUsesNativeAOT)" />
+      <WorkloadPublishProperties Include="Interp" Value="$(_WorkloadUsesInterpreter)" />
+      <WorkloadPublishProperties Include="LibraryMode" Value="$(_WorkloadUsesLibraryMode)" />
+      <WorkloadPublishProperties Include="HybridGlobalization" Value="$(HybridGlobalization)" />
+      <WorkloadPublishProperties Include="ResolvedRuntimePack" Value="$(_MonoWorkloadRuntimePackPackageVersion)" />
+    </ItemGroup>
+
+    <AllowEmptyTelemetry EventName="WorkloadPublishProperties" EventData="@(WorkloadPublishProperties)" />
+  </Target>
+
+  <Target Name="_WorkloadGatherTelemetryData">
+    <PropertyGroup>
+      <_WorkloadUsesBlazorWasm>$(UsingMicrosoftNETSdkBlazorWebAssembly)</_WorkloadUsesBlazorWasm>
+      <_WorkloadUsesWasmSDK>$(UsingMicrosoftNETSdkWebAssembly)</_WorkloadUsesWasmSDK>
+      <_WorkloadUsesMonoAOT>$(RunAOTCompilation)</_WorkloadUsesMonoAOT>
+      <_WorkloadUsesMonoAOT Condition="'$(RunAOTCompilation)' == '' and '$(PublishAot)' != 'true' and ('$(TargetPlatformIdentifier)' == 'maccatalyst' or '$(TargetPlatformIdentifier)' == 'ios' or '$(TargetPlatformIdentifier)' == 'tvos')">$(_RunAotCompiler)</_WorkloadUsesMonoAOT>
+      <_WorkloadUsesNativeAOT>$(PublishAot)</_WorkloadUsesNativeAOT>
+      <_WorkloadUsesInterpreter>$(MonoForceInterpreter)</_WorkloadUsesInterpreter>
+      <_WorkloadUsesInterpreter Condition="'$(_WorkloadUsesInterpreter)' == '' and '$(UseInterpreter)' == 'true'">true</_WorkloadUsesInterpreter>
+      <_WorkloadUsesInterpreter Condition="'$(_WorkloadUsesInterpreter)' == '' and '$(RunAOTCompilation)' != 'true' and ('$(_WorkloadUsesBlazorWasm)' == 'true' or '$(_WorkloadUsesWasmSDK)' == 'true')">true</_WorkloadUsesInterpreter>
+      <_WorkloadUsesLibraryMode Condition="'$(NativeLib)' != '' and ('$(_WorkloadUsesMonoAOT)' == 'true' or '$(_WorkloadUsesNativeAOT)' == 'true')">true</_WorkloadUsesLibraryMode>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <_WorkloadUsesAvalonia Condition="'$([System.IO.Path]::GetFileName(%(ReferencePath.Identity)).ToLower())' == 'avalonia.dll'">true</_WorkloadUsesAvalonia>
```

Comment:
would checking the PackageReference work too?

Reply:
In an avalonia ios sample, it'll have Avalonia.iOS as the PackageReference. I think it's simpler to use the referencepath since there's a consistent avalonia.dll.